lib/view_component/compiler.rb

```diff
@@ -146,21 +146,26 @@ def matching_views_in_source_location
       source_location = component_class.source_location
       return [] unless source_location
 
-      location_without_extension = source_location.chomp(File.extname(source_location))
-
       extensions = ActionView::Template.template_handler_extensions.join(",")
 
-      # view files in the same directory as the component
-      sidecar_files = Dir["#{location_without_extension}.*{#{extensions}}"]
-
       # view files in a directory named like the component
       directory = File.dirname(source_location)
       filename = File.basename(source_location, ".rb")
       component_name = component_class.name.demodulize.underscore
 
+      sub_component_files = if component_class.name.include?("::")
+        subcomponent_path = component_class.name.split("::")[0...-1].map(&:underscore).join("/")
```

Comment:
```suggestion
        subcomponent_path = component_class.name.deconstantize.underscore`
```

but given that we combine `subcomponent_path` with `component_name` that combination is just `component_class.name.underscore`?

Reply:
This is much better, thank you for the suggestion!